### PR TITLE
ci(Pipeline): Refactor icon library build

### LINKED ISSUE #1105: (Spike) Optimise CI pipeline
Our current CI pipeline does a fantastic job of testing and safeguarding our code.

Currently, its average running time is between 10-20 minutes. Whilst this isn't overly long for what we're managing to achieve with it, it does fuel the "rebase queue" issue, where multiple PRs take a long time to merge due to the CI tool running.

This ticket is completed when we have reviewed the current CI pipeline and investigated if/how we can improve it.

Initial areas to focus on:
- parallelisation of build tasks
- checking for duplicate tasks*

(*) The Icon Library build task is run in at least two pipelines, often back to back in the same PR - this is definitely a candidate for optimisation.

### PR DESCRIPTION
**Pull request by elglup**
## Related issue

Fixes #1105 

## Overview
Optimising CI pipelines

## Reason

To review and streamline CI pipelines where possible

## Work carried out

Edit CI config
Added new job to build icon library.  Artifacts are then shared to remaining jobs in pipeline:
test_react_component-library
test_visual_regression
test_docs-site
publish_next
publish_latest

Workflows updated:
build_and_test_feature
build_test_and_deploy_next
build_test_and_deploy_latest

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -289,13 +328,15 @@ workflows:
...
           requires:
+            - build_icon-library
             - lint_react-component-library
       - test_visual_regression:
           requires:
             - test_react-component-library
       - publish_latest:
           requires:
-            - test_visual_regression
+            - test_visual_regression
```
Comment (thyhjwb6) [inline .circleci/config.yml]: There's a newline missing here

**Review by thyhjwb6 — COMMENTED** [review]
Just a new line missing at the bottom of the config

**Review by m7kvqbe1 — APPROVED** [review]
LGTM - just the new line eof thing

**Comment by m7kvqbe1** [comment]
If you are using VSCode there is a plugin called editorconfig (https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) that will make the editor automatically add the new lines for you on save.

It respects the configuration included in the repo in this file here:

https://github.com/Royal-Navy/design-system/blob/master/.editorconfig

There should also be plugins for other IDEs like WebStorm too.

File: .circleci/config.yml (hunk 1 above)
Comment (elglup) [reply .circleci/config.yml]: Thanks, fixed the newline now

**Pull request merged**